### ISSUE TITLE
Lock version

### ISSUE THREAD

**Issue by OliverJAsh** [jspm/jspm-cli#743]
The SystemJS/ES6 Module Loader versions are not locked in the SystemJS config file. This means that when you run `jspm install`, you may end up accidentally upgrading these dependencies without realising. Would it be a good idea to add these to the SystemJS config?

**Comment by guybedford** [comment]
@OliverJAsh the version of SystemJS and Module Loader directly corresponds to the version of jspm. We need to have that synchronisation. So the way to lock the module loaders is to lock the version of jspm. That said, the patch versions of the loaders aren't locked in this way, but loader patches shouldn't downloaded without explicitly running `jspm dl-loader` or upgrading jspm.

**Comment by OliverJAsh** [comment]
Thanks for the information, @guybedford. /cc @rich-nyugen

**Comment by OliverJAsh** [comment]
> That said, the patch versions of the loaders aren't locked in this way, but loader patches shouldn't downloaded without explicitly running jspm dl-loader or upgrading jspm.

If we locked the patch versions in the System config (or elsewhere), as we do for other dependencies, it means that everyone running the same code will be using the same patch version. Likewise, we can commit code to switch between patch versions. This is sometimes necessary to support specific versions.

**Comment by guybedford** [comment]
@OliverJAsh are there ever cases where you'd want the patch of SystemJS to be backtracked? Or is this specifically about ensuring everyone is on the same upgraded version for new features? Because we could better push out updates of SystemJS. 

The issue with locking the version is that it isn't a normal dependency here.

**Comment by OliverJAsh** [comment]
@guybedford There could be cases where you would want that, but it is also about the latter.

I'll have a think about how we can support this.

**Comment by guybedford** [comment]
Each release of jspm comes with a SystemJS version range intended, so perhaps what you're really asking for is more regular releases to ensure upgrades here for beta users?

**Comment by OliverJAsh** [comment]
This has just caught us out. Our builds run `jspm dl-loader` which meant SystemJS will update to any new patch versions. A change in SystemJS has caused our tests to start failing, and we can't downgrade to the old SystemJS.

**Comment by OliverJAsh** [comment]
For some reason, `System.import` promises seems slower in the new release, which causes our tests to timeout on slower machines. Any ideas why this might be?

**Comment by guybedford** [comment]
That's really interesting. Would be great to catch any perf regression - but it would be specific to the type of test being run. Let me know if you can share any more info at all.

As for the version locking, how about a `jspm dl-loader -v 0.18.4` for downloading exact versions of SystemJS?

**Comment by OliverJAsh** [comment]
> As for the version locking, how about a jspm dl-loader -v 0.18.4 for downloading exact versions of SystemJS?

That sounds ideal. jspm would require a minor version, but allow you to change the patch?

**Comment by OliverJAsh** [comment]
> Would be great to catch any perf regression

Hmm, we just had code like this in our tests:

``` js
beforeEach(function (done) { System.import(['foo', 'bar']).then(done); });
```

The default timeout was 5 seconds. For some reason, after the update, this code no longer completed within the default timeout on our test boxes. (Locally, it did, but ~1/10 times it would fail.)

**Comment by guybedford** [comment]
Specifically I mean if you have any idea what sort of module formats / trees / configurations you are importing with that are slower. If it's just the whole test suite, it would be really interesting to run it against a few builds of SystemJS to see if anything in particular stands out. Not getting any slow down in sources here, and the main thing we don't have is large source files. So I'm wondering if it is in the regular expression changes in SystemJS applied to large sources or something like that (https://github.com/systemjs/systemjs/issues/629, https://github.com/systemjs/systemjs/commit/cff37012c238cd949d658f8e6057111d21a6d61a).

**Comment by guybedford** [comment]
Contributions welcome!

**Comment by guybedford** [comment]
Btw, the performance issue is being checked in https://github.com/systemjs/systemjs/issues/653 currently.

**Comment by guybedford** [comment]
Released in jspm@0.16.0-beta.6.
